### PR TITLE
Rewrites FakeShoeboxServiceClient without database => Removes database dependencies from Search Tests

### PR DESCRIPTION
**Pull request by leogrim**
-- technical debt

### CODE REVIEW

File: kifi-backend/modules/common/test/com/keepit/shoebox/FakeShoeboxServiceClientImpl.scala [hunk 1]
```diff
@@ -1,103 +1,230 @@
...
-     Promise.successful(userOpt).future
+    val userOpt =  allUserExternalIds.get(id)
+    Promise.successful(userOpt).future
   }
 
   def getUser(id: Id[User]): Future[User] = {
-    //call(routes.ShoeboxController.getUser(id)).map(r => UserSerializer.userSerializer.reads(r.json))
-    ???
+    val user = allUsers(id)
+    Promise.successful(user).future
```
Comment (gmethvin) [inline kifi-backend/modules/common/test/com/keepit/shoebox/FakeShoeboxServiceClientImpl.scala]: `Future.successful(user)`

**Comment by gmethvin** [comment]
Nice!
-- technical debt

File: kifi-backend/modules/common/test/com/keepit/shoebox/FakeShoeboxServiceClientImpl.scala [hunk 2]
```diff
@@ -1,104 +1,231 @@
...
-)
-    extends ShoeboxServiceClient {
+class FakeShoeboxServiceClientImpl(clickHistoryTracker: ClickHistoryTracker, browsingHistoryTracker: BrowsingHistoryTracker) extends ShoeboxServiceClient {
   val host: String = ""
   protected def httpClient: com.keepit.common.net.HttpClient = ???
 
+  // Fake ID counters
+
+  private val userIdCounter = new AtomicInteger(0)
+  private def nextUserId = { Id[User](userIdCounter.incrementAndGet()) }
```
Comment (gmethvin) [inline kifi-backend/modules/common/test/com/keepit/shoebox/FakeShoeboxServiceClientImpl.scala]: I think these should have parens

**Comment by yingjieMiao** [comment]
Looks great!

File: kifi-backend/modules/common/test/com/keepit/search/SearchConfigTest.scala [hunk 3]
```diff
@@ -33,29 +28,25 @@ class SearchConfigTest extends Specification with DbRepos {
...
       running(new EmptyApplication().withFakePersistEvent.withShoeboxServiceModule) {
+        val fakeShoeboxServiceClient = inject[ShoeboxServiceClient].asInstanceOf[FakeShoeboxServiceClientImpl]
         val searchConfigManager = new SearchConfigManager(None, inject[ShoeboxServiceClient], inject[MonitoredAwait])
-        val searchConfigExperimentRepo = inject[SearchConfigExperimentRepo]
-        val userRepo = inject[UserRepo]
 
-        val andrew = db.readWrite { implicit s =>
-          userRepo.save(User(firstName = "Andrew", lastName = "Connor"))
-        }
+        val Seq(andrew) = fakeShoeboxServiceClient.saveUsers(User(firstName = "Andrew", lastName = "Connor"))
```
Comment (eishay) [inline kifi-backend/modules/common/test/com/keepit/search/SearchConfigTest.scala:34]: nice, forgot you can do it with scala!

**Pull request merged**